pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -2034,6 +2034,235 @@ func TestApplyParameters_ArrayIndexing(t *testing.T) {
 	}
 }
 
+func TestApplyReplacementsMatrix(t *testing.T) {
+	ctx := context.Background()
+	cfg := config.FromContextOrDefaults(ctx)
+	cfg.FeatureFlags.EnableAPIFields = config.BetaAPIFields
+	ctx = config.ToContext(ctx, cfg)
+	for _, tt := range []struct {
+		name     string
+		original v1beta1.PipelineSpec
+		params   []v1beta1.Param
+		expected v1beta1.PipelineSpec
+	}{{
+		name: "matrix params replacement",
```

Comment:
Are there existing tests for this functionality? (I hope so!) Same question for "matrix params with final tasks"